From 9c921b9515ad55583535bcca389dc98f4070c6b1 Mon Sep 17 00:00:00 2001
From: "Signed-off-by: Biju Das" <biju.das@bp.renesas.com>
Date: Thu, 2 Apr 2015 11:42:52 +0100
Subject: [PATCH 1/7] ARM: shmobile: porter board: Remove Audio platform code

No longer used as audio setup is done via Device Tree.

This change is identical to the koelsch patch "ARM: shmobile: Koelsch board: Remove Audio platform code"
from Phil Edworthy

Signed-off-by: Biju Das <biju.das@bp.renesas.com>
---
 arch/arm/mach-shmobile/board-porter-reference.c |   79 -----------------------
 1 file changed, 79 deletions(-)

diff --git a/arch/arm/mach-shmobile/board-porter-reference.c b/arch/arm/mach-shmobile/board-porter-reference.c
index 43c262e..ae06151 100644
--- a/arch/arm/mach-shmobile/board-porter-reference.c
+++ b/arch/arm/mach-shmobile/board-porter-reference.c
@@ -126,78 +126,6 @@ static void __init porter_add_du_device(void)
 	platform_device_register_full(&info);
 }
 
-/* Sound */
-static struct rsnd_ssi_platform_info rsnd_ssi[] = {
-	RSND_SSI(AUDIOPP_DMAC_SLAVE_CMD0_TO_SSI0, gic_spi(370), 0),
-	RSND_SSI(AUDIOPP_DMAC_SLAVE_SSI1_TO_SCU1, gic_spi(371), RSND_SSI_CLK_PIN_SHARE),
-};
-
-static struct rsnd_src_platform_info rsnd_src[2] = {
-	RSND_SRC(0, AUDIO_DMAC_SLAVE_SCU0_TX, gic_spi(352)),
-	RSND_SRC(0, AUDIO_DMAC_SLAVE_CMD1_TO_MEM, gic_spi(353)),
-};
-
-static struct rsnd_dvc_platform_info rsnd_dvc[2] = {
-};
-
-static struct rsnd_dai_platform_info rsnd_dai = {
-	.playback = { .ssi = &rsnd_ssi[0], .src = &rsnd_src[0], .dvc = &rsnd_dvc[0], },
-	.capture  = { .ssi = &rsnd_ssi[1], .src = &rsnd_src[1], .dvc = &rsnd_dvc[1], },
-};
-
-static struct rcar_snd_info rsnd_info = {
-	.flags		= RSND_GEN2,
-	.ssi_info	= rsnd_ssi,
-	.ssi_info_nr	= ARRAY_SIZE(rsnd_ssi),
-	.src_info	= rsnd_src,
-	.src_info_nr	= ARRAY_SIZE(rsnd_src),
-	.dvc_info	= rsnd_dvc,
-	.dvc_info_nr	= ARRAY_SIZE(rsnd_dvc),
-	.dai_info	= &rsnd_dai,
-	.dai_info_nr	= 1,
-};
-
-static struct asoc_simple_card_info rsnd_card_info = {
-	.name		= "SSI01-AK4643",
-	.codec		= "ak4642-codec.2-0012",
-	.platform	= "rcar_sound",
-	.daifmt		= SND_SOC_DAIFMT_I2S | SND_SOC_DAIFMT_CBS_CFS,
-	.cpu_dai = {
-		.name	= "rcar_sound",
-	},
-	.codec_dai = {
-		.name	= "ak4642-hifi",
-		.sysclk	= 11289600,
-	},
-};
-
-static void __init porter_add_rsnd_device(void)
-{
-	struct resource rsnd_resources[] = {
-		[RSND_GEN2_SCU]  = DEFINE_RES_MEM(0xec500000, 0x1000),
-		[RSND_GEN2_ADG]  = DEFINE_RES_MEM(0xec5a0000, 0x100),
-		[RSND_GEN2_SSIU] = DEFINE_RES_MEM(0xec540000, 0x1000),
-		[RSND_GEN2_SSI]  = DEFINE_RES_MEM(0xec541000, 0x1280),
-	};
-
-	struct platform_device_info cardinfo = {
-		.parent         = &platform_bus,
-		.name           = "asoc-simple-card",
-		.id             = -1,
-		.data           = &rsnd_card_info,
-		.size_data      = sizeof(struct asoc_simple_card_info),
-		.dma_mask       = DMA_BIT_MASK(32),
-	};
-
-	platform_device_register_resndata(
-		&platform_bus, "rcar_sound", -1,
-		rsnd_resources, ARRAY_SIZE(rsnd_resources),
-		&rsnd_info, sizeof(rsnd_info));
-
-	platform_device_register_full(&cardinfo);
-}
-
-
 /*
  * This is a really crude hack to provide clkdev support to platform
  * devices until they get moved to DT.
@@ -209,12 +137,6 @@ static const struct clk_name clk_names[] __initconst = {
 	{ "du1", "du.1", "rcar-du-r8a7791" },
 	{ "lvds0", "lvds.0", "rcar-du-r8a7791" },
 	{ "hsusb", NULL, "usb_phy_rcar_gen2" },
-	{ "ssi0", "ssi.0", "rcar_sound" },
-	{ "ssi1", "ssi.1", "rcar_sound" },
-	{ "src0", "src.0", "rcar_sound" },
-	{ "src1", "src.1", "rcar_sound" },
-	{ "dvc0", "dvc.0", "rcar_sound" },
-	{ "dvc1", "dvc.1", "rcar_sound" },
 	{ "vin0", NULL, "r8a7791-vin.0" },
 	{ "vsps", NULL, NULL },
 #if IS_ENABLED(CONFIG_VIDEO_RENESAS_VSP1)
@@ -927,7 +849,6 @@ static void __init porter_add_standard_devices(void)
 
 	porter_add_du_device();
 	porter_add_usb_devices();
-	porter_add_rsnd_device();
 	porter_add_camera0_device();
 #if IS_ENABLED(CONFIG_VIDEO_RENESAS_VSP1)
 	porter_add_vsp1_devices();
-- 
1.7.9.5

